### PR TITLE
samples(compute): add disks samples

### PR DESCRIPTION
**Pull request by FrodoTheTrue**

### CODE REVIEW

**Comment by codyoss** [comment]
@quartzmo Would you review this PR

File: compute/disks/set_disk_autodelete.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	"context"
+	"fmt"
+	"io"
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+)
+
+// setDiskAutodelete sets the autodelete flag of a disk to given value as true.
+func setDiskAutodelete(
```
[the remark this answers is not in the document]
Comment (quartzmo) [reply compute/disks/set_disk_autodelete.go:28]: > sets the autodelete flag

Use correct capitalization for `AutoDelete` field?

File: compute/disks/create_empty_disk.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// createEmptyDisk creates a new empty disk in a project in given zone.
+func createEmptyDisk(
+	w io.Writer,
+	projectID, zone, diskName, diskType string,
```
Comment (quartzmo) [inline compute/disks/create_empty_disk.go:31]: Why is `SizeGb` omitted from these params?

File: compute/disks/create_disk_from_snapshot.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+	compute "cloud.google.com/go/compute/apiv1"
+	computepb "google.golang.org/genproto/googleapis/cloud/compute/v1"
+	"google.golang.org/protobuf/proto"
+)
+
+// createDiskFromSnapshot creates a new disk in a project in given zone.
+func createDiskFromSnapshot(
+	w io.Writer,
+	projectID, zone, diskName, diskType, snapshotLink string,
```
Comment (quartzmo) [inline compute/disks/create_disk_from_snapshot.go:31]: Why is `SizeGb` omitted from these params?

**Comment by FrodoTheTrue** [comment]
Regarding renaming of the folder - I would prefer to left `disks`, to be consistent with python structure: https://github.com/googleapis/python-compute/tree/main/samples/snippets

File: compute/disks/disks_test.go [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+
+	t.Run("createDiskSnapshot and deleteDisk", func(t *testing.T) {
+		err := createDisk(ctx, tc.ProjectID, zone, diskName, sourceImage)
+		if err != nil {
+			t.Fatalf("createDisk got err: %v", err)
+		}
+
+		err = createDiskSnapshot(ctx, tc.ProjectID, zone, diskName, snapshotName)
+		if err != nil {
+			t.Errorf("createDiskSnapshot got err: %v", err)
```
[the remark this answers is not in the document]
Comment (FrodoTheTrue) [reply compute/disks/disks_test.go:167]: but if I understood correctly in this case will not delete the disk created here: https://github.com/GoogleCloudPlatform/golang-samples/pull/2628/files/3ed7310ea030f23ddc9d747772f3b9541a4bf590#diff-8c1d3e8777449a285bf474a434998f576b0a37604c242aded58ff842441448cdR160

File: compute/disks/create_empty_disk.go (hunk 2 above)
Comment (FrodoTheTrue) [reply compute/disks/create_empty_disk.go:31]: We decided previously to hard code it: https://github.com/GoogleCloudPlatform/golang-samples/pull/2623#discussion_r915064713

**Comment by FrodoTheTrue** [comment]
@quartzmo as we decided to use fatalf I have just added additional fixes. Could you please review again?

File: compute/disks/disks_test.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (quartzmo) [reply compute/disks/disks_test.go]: OK, I see. Please use `Errorf` if concerned about orphaned disk.

**Review by quartzmo — APPROVED** [review]

**Comment by quartzmo** [comment]
LGTM. Please use `Errorf` after creating disk but before deleting disk if concerned about orphaned disk.

**Comment by quartzmo** [comment]
@codyoss Please provide code owner review, thanks.

**Review by codyoss — APPROVED** [review]

**Pull request merged**